### PR TITLE
update selecting cards with different name/level etc.

### PR DESCRIPTION
**Pull request by mercury233**

### CODE REVIEW

File: c10424147.lua [hunk 1]
```diff
@@ -80,7 +80,7 @@ function c10424147.sptg2(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
 	if chkc then return false end
 	local g=Duel.GetMatchingGroup(c10424147.spfilter3,tp,LOCATION_MZONE+LOCATION_GRAVE,0,nil,e)
 	if chk==0 then return Duel.GetLocationCountFromEx(tp)>0
-		and g:CheckSubGroup(aux.dncheck,3,3)
+		and g:GetClassCount(Card.GetCode)>=3
```
Comment (nekrozar) [inline c10424147.lua]: why?

Comment (mercury233) [reply c10424147.lua]: may have better performance

Comment (nekrozar) [reply c10424147.lua]: however, _Simorgh Onslaught_ hasn't changed(maybe other scripts).

Comment (mercury233) [reply c10424147.lua]: _Simorgh Onslaught_ is updated.

**Pull request merged**